src/patternfly/components/Page/page.scss

```diff
@@ -272,50 +279,51 @@
   margin-bottom: var(--pf-c-page__header-tools--MarginBottom);
   margin-left: auto; // to push it to the right
 
-  .pf-m-icons {
+  // Hides on mobile visible on desktop
+  .pf-m-desktop {
     display: none;
     visibility: hidden;
 
     @media screen and (min-width: $pf-global--breakpoint--lg) {
-      display: block;
+      display: inline-block;
       visibility: visible;
     }
   }
 
+  // Hides on desktop visible on mobile
   .pf-m-mobile {
-    margin-right: 0;
+    display: inline-block;
+    visibility: visible;
 
     @media screen and (min-width: $pf-global--breakpoint--lg) {
       display: none;
       visibility: hidden;
     }
   }
 
-  .pf-m-user {
+  .pf-c-avatar {
     display: none;
+    margin-left: 0;
     visibility: hidden;
 
-    @media screen and (min-width: $pf-global--breakpoint--md) {
+    @media screen and (min-width: $pf-global--breakpoint--lg) {
+      margin-left: var(--pf-c-page__header-tools--c-avatar--MarginLeft);
```

Comment:
couldn't you just leave this on `.pf-c-avatar` since it won't apply when it's `display: none;`?